packages/neon/neon/lib/src/widgets/app_bar.dart

```diff
@@ -68,19 +68,103 @@ class NeonAppBar extends StatelessWidget implements PreferredSizeWidget {
               ],
             ],
           ),
-          actions: [
-            const NotificationIconButton(),
-            IconButton(
-              onPressed: () {
-                AccountSettingsRoute(accountid: account.id).go(context);
-              },
-              tooltip: AppLocalizations.of(context).settingsAccount,
-              icon: NeonUserAvatar(
-                account: account,
+          actions: const [
+            NotificationIconButton(),
+            AccountSwitcherButton(),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+@internal
+class AccountSwitcherButton extends StatelessWidget {
+  const AccountSwitcherButton({
+    super.key,
+  });
+
+  Future<void> _onPressed(final BuildContext context) async {
+    final accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    final accounts = accountsBloc.accounts.value;
+    final aa = accountsBloc.activeAccount.value!;
+
+    await showDialog(
+      context: context,
+      builder: (final context) {
+        final accountList = <Widget>[];
+
+        for (final account in accounts) {
+          if (account.id == aa.id) {
+            continue;
+          }
+
+          final tile = NeonAccountTile(
+            account: account,
+            onTap: () {
+              accountsBloc.setActiveAccount(account);
+              Navigator.of(context).pop();
+            },
+          );
+
+          accountList.add(tile);
+        }
+
+        final body = Column(
+          children: [
+            NeonAccountTile(
+              account: aa,
+              trailing: const Icon(Icons.check_circle),
+              onTap: Navigator.of(context).pop,
+            ),
+            const Divider(),
```

Comment:
This divider is a bit useless imo. The active account is already indicated by the check mark.